notebook2/notebook/notebook.py

```diff
@@ -170,61 +198,201 @@ def external_url_for(path):
+            maxDate = condition.last_transition_time
+
+    return {
+        "message": maxCond.message,
+        "reason": maxCond.reason,
+        "status": maxCond.status,
+        "type": maxCond.type}
+
+
+def marshall_notebooks(pods = [], svc_status = None):
+    notebooks = []
+
+    for pod in pods:
+        notebook = {
+            'name': pod.metadata.labels['name'],
+            'pod_name': pod.metadata.name,
+            'svc_name': pod.metadata.labels['svc_name'],
+            'pod_status': pod.status.phase,
+            'creation_date': pod.metadata.creation_timestamp.strftime('%D'),
+            'jupyter_token': pod.metadata.labels['jupyter_token'],
+            'container_status': read_containers_status(pod.status.container_statuses),
+            'condition': read_conditions(pod.status.conditions)
+        }
+
+        notebook['url'] = f"/instance/{notebook['svc_name']}/?token={notebook['jupyter_token']}"
+
+        if svc_status is not None:
+            notebook['svc_status'] = svc_status
+        else:
+            notebook['svc_status'] = read_svc_status(notebook['svc_name'])
+
+        notebooks.append(notebook)
+
+    return notebooks
+
+
+def get_live_user_notebooks(user_id = None):
+    if user_id is None:
+        return None
+
+    pods = k8s.list_namespaced_pod(
+        namespace='default',
+        label_selector=f"user_id={user_id}",
+        timeout_seconds=KUBERNETES_TIMEOUT_IN_SECONDS).items
+
+    if len(pods) == 0:
+        return None
+
+    # Kubernetes service will reflect the change immediately, pod will not
+    notebooks = list(filter(lambda n: n['svc_status'] != 'Deleted', marshall_notebooks(pods)))
+
+    if len(notebooks) == 0:
+        return None
+
+    return notebooks
+
+
 @app.route('/healthcheck')
 def healthcheck():
     return '', 200
 
 
-@app.route('/')
+@app.route('/', methods=['GET'])
 def root():
-    if 'svc_name' not in session:
-        log.info(f'no svc_name found in session {session.keys()}')
-        return render_template('index.html',
-                               form_action_url=external_url_for('new'),
+    return render_template('index.html')
+
+
+@app.route('/notebook', methods=['GET'])
+@requires_auth()
+def notebook_page():
+    notebooks = get_live_user_notebooks(user_id = UNSAFE_user_id_transform(session['user']['id']))
+
+    if notebooks is None:
+        return render_template('notebook.html',
+                               form_action_url=flask.url_for('notebook_post'),
                                images=list(WORKER_IMAGES),
                                default='hail')
-    svc_name = session['svc_name']
-    jupyter_token = session['jupyter_token']
-    log.info('redirecting to ' + external_url_for(f'instance/{svc_name}/?token={jupyter_token}'))
-    return redirect(external_url_for(f'instance/{svc_name}/?token={jupyter_token}'))
 
+    # TODO: We currently simplify state tracking, since using vanilla js makes this trickier
```

Comment:
Issue: https://github.com/hail-is/hail/issues/5487

Reply:
we should `assert len(notebooks) == 1` and comment the issue number